### PR TITLE
Add the ability for users to create admin comments on requests...

### LINKED ISSUE #519: Add staff note field to request mediation space
Staff in Branner have requested an area in the staff mediation space for Branner Maps to record notes on a particular request.  They would use the notes field to record and easily communicate the logic behind a decision to approve or not approve a request.  While they didn't specify if the notes should be at the request level or the item level, I think it would have to be at the item level.  Any request based notes could just be recorded on the first item that is listed.

### PR DESCRIPTION
**Pull request by jkeck**
...they can manage.

Closes #519 

admin-comments-spec

### CODE REVIEW

File: app/assets/javascripts/adminComments.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+var adminComments = (function() {
```
Comment (jkeck) [inline app/assets/javascripts/adminComments.js]: This is used in another javascript library, but I guess Hound doesn't catch that.

File: app/assets/javascripts/adminComments.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  var adminCommentToggleSelector = '[data-behavior="toggle-admin-comment-form"]';
+  var adminCommentFormCancelButton = '[data-behavior="admin-comment-cancel"]';
+
+  function commentTemplate(comment) {
+    return '<li>' +
+             comment.comment +
+             '<span class="text-muted">' +
+               ' - ' +
+               comment.commenter + ' - ' +
+               comment.created_at +
```
Comment (jkeck) [inline app/assets/javascripts/adminComments.js]: This comes from the active record attribute name, so I think we should keep it.

File: spec/requests/admin_comments_requests_spec.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  let(:headers) { { 'HTTP_REFERER' => 'http://example.com' } }
+  let(:url) { "/mediated_pages/#{mediated_page.id}/admin_comments" }
+
+  before do
+    stub_current_user(user)
+  end
+
+  describe '#create' do
+    context 'when successful' do
+      it "merges the current user's webauth into the persisten comment as the commenter" do
```
Comment (jmartin-sul) [inline spec/requests/admin_comments_requests_spec.rb]: typo? (`persisten`)

File: app/assets/javascripts/adminComments.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      this.onAjaxError();
+
+      this.addCancelButtonBehavior();
+    },
+
+    addCommentToggleBehavior: function() {
+      var _this = this;
+      var commentToggle = _this.holdingsTable.find(adminCommentToggleSelector);
+      commentToggle.on('click', function() {
+        _this.form.slideToggle();
```
Comment (jkeck) [inline app/assets/javascripts/adminComments.js]: D'oh! Totally what I was intending by moving that into its own function.  Good catch!

**Comment by jkeck** [comment]
@jmartin-sul good :eye:! I think I have addressed the issues you pointed out.

**Pull request merged**